integreat_cms/cms/views/pages/page_form_view.py

```diff
@@ -288,29 +288,28 @@ def post(self, request, *args, **kwargs):
                 page_translation_form.instance.page.translations.filter(
                     language__in=languages
                 ).update(status=status.DRAFT)
-            # If the translation is minor change and there is no major public version
-            if (page_translation_form.instance.minor_edit is True) and (
-                not page_translation_form.instance.major_public_version
+            # If this is the first version and the minor edit checkbox is checked, remove it
+            if (
+                page_translation_form.instance.version == 1
+                and page_translation_form.instance.minor_edit
             ):
-                # and if it is the first version, save it as major public
-                if page_translation_form.instance.version == 1:
-                    page_translation_form.instance.minor_edit = False
-                    page_translation_form.save()
-                    messages.info(
-                        request,
-                        _(
-                            'Page "{}" was saved as major version. The first version is not allowed to be a minor edit.'
-                        ).format(page_translation_form.instance.title),
-                    )
-                # or if there are only draft and/or minor public version, change them to public major version
-                else:
-                    language_tree_node = region.language_node_by_slug.get(language.slug)
-                    languages = [language] + [
-                        node.language for node in language_tree_node.get_descendants()
-                    ]
-                    page_translation_form.instance.page.translations.filter(
-                        language__in=languages
-                    ).update(status=status.PUBLIC)
+                page_translation_form.instance.minor_edit = False
+                page_translation_form.instance.save()
+                messages.info(
+                    request,
+                    _(
+                        'The "minor edit" option was disabled because the first version can never be a minor edit.'
+                    ).format(page_translation_form.instance.title),
+                )
+            # When a version is published and a minor edit, also publish the past versions
+            # (to make sure the translation state is updated correctly)
+            if (
```

Comment:
As of discussion in https://github.com/digitalfabrik/integreat-cms/issues/1464 this feels wrong to me, because we're mixing the intention of 2 settings. But if everyone feels this is good, then I won't object.

I personally feel we should definitely untangle the draft/public state from minor edits (ignore in translation status).

Reply:
But maybe we need to rephrase the concept of the translation status. And maybe that is something that only needs to happen in my head. But if we label the translation status to "Translation status of public version", then this sure makes sense. Or maybe we just need to differentiate between these to options: we can have public up to date translations and we can have draft up to date translations.

Reply:
This probably needs separate clarification with the service team. But in the mean time, I guess we should leave the current behaviour (that draft pages are considered missing) because this is what we teach our users at the moment.

Reply:
But in general I agree that there is probably a better solution out there :smile:

Reply:
Okay, let's continue this discussion after merging ^^